src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -327,6 +331,71 @@ class ParseLiveQueryServer {
       if (isSubscriptionSessionTokenMatched) {
         return Parse.Promise.as(true);
       }
+
+      // Check if the user has any roles that match the ACL
+      return new Parse.Promise(function(resolve, reject){
+
+        // resolve false right away if the acl doesn't have any roles
+        var acl_has_roles = false;
+        for(var key in acl.permissionsById) {
+            if(key.startsWith("role:")) {
+                acl_has_roles = true;
+                break;
+            }
+        }
+        if(!acl_has_roles) {
+            return resolve(false);
+        }
+
+        _this.sessionTokenCache.getUserId(subscriptionSessionToken)
+        .then(function(userId){
```

Comment:
since we're using babel, I couldn't resist trying to imutableize this ;).  Not asking you to change it, but since I went through the effort to test it, I figured I'd put it here for posterity's sake:

```
return new Parse.Promise(function (resolve, reject) {
    // resolve false right away if the acl doesn't have any roles
    const acl_has_roles = Object.keys(acl.permissionsById)
        .some(key => key.startsWith("role:"));
    if(!acl_has_roles) {
        return resolve(false);
    }
...
```